aws/src/main/java/org/apache/iceberg/aws/s3/VendedCredentialsProvider.java

```diff
@@ -58,6 +62,8 @@ private VendedCredentialsProvider(Map<String, String> properties) {
         CachedSupplier.builder(() -> credentialFromProperties().orElseGet(this::refreshCredential))
             .cachedValueName(VendedCredentialsProvider.class.getName())
             .build();
+    this.catalogEndpoint = properties.get(CatalogProperties.URI);
```

Comment:
can you please update `Preconditions.checkArgument(null != properties.get(URI), "Invalid URI: null");` to `Preconditions.checkArgument(null != properties.get(URI), "Invalid credentials URI: null");` and also add `Preconditions.checkArgument(null != properties.get(CatalogProperties.URI), "Invalid catalog URI: null");`.

Please also update `invalidOrMissingUri()` and add a check where the catalog URI isn't provided